### PR TITLE
Fix class weight labels

### PR DESCRIPTION
**Pull request by TheBicPen**
#### Reference Issues/PRs
Fixes #22413

#### What does this implement/fix? Explain your changes.
Let `y` determine the classes instead of `class_weight`.

#### Any other comments?
I had some issues running the tests - import errors when running `pytest sklearn/utils/tests/test_class_weight.py`. Running just my tests worked fine though.

### CODE REVIEW

**Comment by TheBicPen** [comment]
Ready for review

**Comment by AbChatt** [comment]
Reviewed, everything looks good!

**Review by tahmid-haque — CHANGES_REQUESTED** [review]
Code itself looks good and it looks like a few test cases fail (that weren't failing before). I think you should take a look to confirm. I ran `pytest sklearn/utils/tests/test_class_weight.py` and 3 tests: `test_compute_class_weight_not_present`, `test_compute_class_weight_dict`, `test_compute_class_weight_default` failed.

**Comment by TheBicPen** [comment]
>I ran `pytest sklearn/utils/tests/test_class_weight.py` 
I'm getting a `ModuleNotFoundError` when I try to run that command.

**Comment by tahmid-haque** [comment]
> > I ran `pytest sklearn/utils/tests/test_class_weight.py`
> 
> I'm getting a `ModuleNotFoundError` when I try to run that command.

I had that issue as well and it prevented me from running any of the tests. Try following [this](https://scikit-learn.org/stable/developers/advanced_installation.html#install-bleeding-edge). It works after I followed the steps.

**Comment by TheBicPen** [comment]
Looks like there were some upstream changes to some cython modules, which required recompilation. Looking into the failed test cases now.

**Review by tahmid-haque — APPROVED** [review]
Looks good!
